core/src/main/java/com/amplifyframework/predictions/models/BoundedKeyValue.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.models;
+
+import androidx.annotation.NonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Holds the key-value detection results
+ * for the predictions category.
+ */
+public final class BoundedKeyValue extends ImageFeature<Map.Entry<String, String>> {
```

Comment:
It would be better not to use `Map.Entry`, here, unless you're working with a `Map`.

Android's `Pair` could be a candidate: https://developer.android.com/reference/android/util/Pair

Actually, you could have this just be `BoundedKeyValue<K,V>`, and then pass the key and vaule types into `ImageFeature`, somehow?

Wait, but also - why does a bounded key value pair extend an "Image" thing?

Reply:
It seems to be something that is returned by Image identification method. Not sure what it means to detect a key-value pair from a document image though...